cmd/create/machinepool/nodepool.go

```diff
@@ -33,6 +33,12 @@ func addNodePool(cmd *cobra.Command, clusterKey string, cluster *cmv1.Cluster, r
 		os.Exit(1)
 	}
 
+	isMultiAvailabilityZoneSet := cmd.Flags().Changed("multi-availability-zone")
+	if isMultiAvailabilityZoneSet && cluster.Managed() {
```

Comment:
The flow under the nodepool will happen only on Hosted Control Planes.

In the cmd you'll see:

```
	if cluster.Hypershift().Enabled() {
		addNodePool(cmd, clusterKey, cluster, r)
	} else {
		addMachinePool(cmd, clusterKey, cluster, r)
	}
```

So, no need to add the "cluster.Managed()".

Perhaps the same validation should be also added in the edit command.